### PR TITLE
ApplicationApiRoute bound to Utxo model

### PR DESCRIPTION
**Pull request by alsala**
`ApplicationApiRoute` and `ApplicationApiGroup` classes are strictly bound to Utxo model, but used both in Utxo and Account models. This class is made generic and an account model specific implementation is provided as well.

https://horizenlabs.atlassian.net/browse/EVM-679

### CODE REVIEW

File: sdk/src/main/scala/io/horizen/account/AccountSidechainApp.scala [hunk 1]
```diff
@@ -148,6 +146,8 @@ class AccountSidechainApp @Inject()
   // Init Sync Status actor
   val syncStatusActorRef: ActorRef = SyncStatusActorRef("SyncStatus", sidechainSettings, nodeViewHolderRef, params, timeProvider)
 
+  override lazy val applicationApiRoutes: Seq[AccountApplicationApiRoute] = customApiGroups.asScala.map(apiRoute => AccountApplicationApiRoute(settings.restApi, apiRoute, nodeViewHolderRef))
```
Comment (i-Alex) [inline sdk/src/main/scala/io/horizen/account/AccountSidechainApp.scala:149]:
```suggestion
  override lazy val applicationApiRoutes: Seq[ApiRoute] = customApiGroups.asScala.map(apiRoute => AccountApplicationApiRoute(settings.restApi, apiRoute, nodeViewHolderRef))
```
I assume that Seq type must remain the same as in the base class.

File: sdk/src/main/scala/io/horizen/account/api/http/route/AccountApplicationApiRoute.scala [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  private def sendMessageToSidechainNodeView[T, R](messageToSend: T): R = {
+    try {
+      val res = (sidechainNodeViewHolderRef ? messageToSend).asInstanceOf[Future[R]]
+      val result = Await.result[R](res, settings.timeout)
+      result
+    }
+    catch {
+      case e: Exception => throw new Exception(e)
+    }
+  }
```
Comment (i-Alex) [inline sdk/src/main/scala/io/horizen/account/api/http/route/AccountApplicationApiRoute.scala:74]: can we put all this methods to the base class?
Why do we have the same code here and in utxo class `SidechainApplicationApiRoute`?

File: sdk/src/main/scala/io/horizen/utxo/SidechainApp.scala [hunk 3]
```diff
@@ -214,6 +213,8 @@ class SidechainApp @Inject()
 
   val boxIterator: BoxIterator = backupStorage.getBoxIterator
 
+  override lazy val applicationApiRoutes: Seq[SidechainApplicationApiRoute] = customApiGroups.asScala.map(apiRoute => SidechainApplicationApiRoute(settings.restApi, apiRoute, nodeViewHolderRef))
```
Comment (i-Alex) [inline sdk/src/main/scala/io/horizen/utxo/SidechainApp.scala:216]:
```suggestion
  override lazy val applicationApiRoutes: Seq[ApiRoute] = customApiGroups.asScala.map(apiRoute => SidechainApplicationApiRoute(settings.restApi, apiRoute, nodeViewHolderRef))
```
same as in Account class

File: sdk/src/test/java/io/horizen/utxo/api/http/SimpleCustomApi.java [hunk 4]
```diff
@@ -34,7 +34,7 @@ public List<Route> getRoutes() {
     private ApiResponse getNSecretsFunction(GetSecretRequest ent) {
         try {
             RespAllSecret resp = new RespAllSecret();
-            List<Secret> res = new ArrayList<>(getFunctionsApplierOnSidechainNodeView().applyFunctionOnSidechainNodeView(v -> v.getNodeWallet().allSecrets()));
+            List<io.horizen.secret.Secret> res = new ArrayList<>(getFunctionsApplierOnSidechainNodeView().applyFunctionOnSidechainNodeView(v -> (v).getNodeWallet().allSecrets()));
```
Comment (i-Alex) [inline sdk/src/test/java/io/horizen/utxo/api/http/SimpleCustomApi.java:37]: why do we need any changes in this line?

**Review by i-Alex — CHANGES_REQUESTED** [review]
Looks good, minor points left

File: sdk/src/main/scala/io/horizen/account/AccountSidechainApp.scala (hunk 1 above)
Comment (alsala) [reply sdk/src/main/scala/io/horizen/account/AccountSidechainApp.scala]: Right, done

File: sdk/src/main/scala/io/horizen/utxo/SidechainApp.scala (hunk 3 above)
Comment (alsala) [reply sdk/src/main/scala/io/horizen/utxo/SidechainApp.scala]: Done

File: sdk/src/test/java/io/horizen/utxo/api/http/SimpleCustomApi.java (hunk 4 above)
Comment (alsala) [reply sdk/src/test/java/io/horizen/utxo/api/http/SimpleCustomApi.java]: No need indeed

**Review by i-Alex — APPROVED** [review]

**Review by saratnt — APPROVED** [review]

**Pull request merged**